bin/ch/ch.cpp

```diff
@@ -831,7 +831,17 @@ int _cdecl wmain(int argc, __in_ecount(argc) LPWSTR argv[])
 #endif
 
 #ifdef _WIN32
-    _setmode(_fileno(stdout), _O_U8TEXT); // enable Unicode output as UTF-8
+    // Set the output mode of stdout so we can display non-ASCII characters on the console and redirect to file as UTF-8
+    {
+        int result = _setmode(_fileno(stdout), _O_U8TEXT); // set stdout to UTF-8 mode
+        if (result == -1)
+        {
+            // Failed to set mode. Undefined behavior may result, so exit now.
+            wprintf(_u("Failed to set output stream mode. Exiting...\n"));
```

Comment:
There's a little bit of irony here that failing to set the mode of stdout might cause undefined behavior when printing to stdout, and then we print an error message to stdout. Here I'm assuming that the mode will remain as the default and ASCII output should work just fine, but if the message comes out corrupted, exiting early was the right decision in any case.